.circleci/config.yml

```diff
@@ -136,391 +145,396 @@ jobs:
-
+            . env/bin/activate
+            make -C docs/ cpp html
+      - store_artifacts:
+          path: ./docs/build/html
       - run:
-          name: create bdist_wheel
+          name: doctest
           command: |
             . env/bin/activate
-            python setup.py bdist_wheel
-
+            make -C docs/ doctest
       - run:
-          name: deploy
+          name: linkcheck
           command: |
             . env/bin/activate
-            python -m pip install twine
-            twine upload -u $PYPI_USERNAME -p $PYPI_PASSWORD --skip-existing ./dist/*
-
-  #
-  # win
-  #
+            make -C docs/ linkcheck
 
-  test-win:
+  test-linux:
     parameters:
       python-version:
         type: string
-      architecture:
+      numpy-version:
+        type: string
+      dimod-version:
         type: string
 
-    executor:
-      name: win/default
-
-    environment:
-      CL: /d2FH4-
+    docker:
+      - image: circleci/python:<< parameters.python-version >>
 
     steps:
       - checkout
-
-      - run: &win-install-python-template
-          name: install python and create virtualenv
+      - attach_workspace:
+          at: dist
+      - run:
+          name: install
           command: |
-            nuget install python<< parameters.architecture >> -Version << parameters.python-version >> -ExcludeVersion -OutputDirectory .
-            .\python<< parameters.architecture >>\tools\python.exe --version
-            .\python<< parameters.architecture >>\tools\python.exe -m pip install virtualenv
-            .\python<< parameters.architecture >>\tools\python.exe -m virtualenv env
-
-      - run: &win-install-dependencies-template
-          name: install dependencies
+            python -m venv env
+            . env/bin/activate
+            pip install 'dimod<< parameters.dimod-version >>' 'numpy<< parameters.numpy-version >>' --only-binary=dimod,numpy
+            pip install dwave-preprocessing --no-index -f dist/ --no-deps --force-reinstall
+      - run: &unix-run-tests
+          name: run tests
           command: |
-            env\Scripts\activate.ps1
-            python --version
+            . env/bin/activate
+            cd tests/
             pip install -r requirements.txt
-            pip install -r tests\requirements.txt
+            python -m unittest
 
-      - run:
-          name: build extension
+  test-linux-cpp11:
+    docker:
+      - image: circleci/python:3.9
+
+    steps:
+      - checkout
+
+      - restore_cache: &restore-cache-env
+          key: v2-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "tests/requirements.txt" }}-{{ checksum "docs/requirements.txt" }}-{{ .Environment.CIRCLE_JOB }}
+
+      - run: &create-virtualenv
+          name: Create virtual environment
           command: |
-            env\Scripts\activate.ps1
-            python setup.py build_ext --inplace
+            python -m pip install virtualenv
+            python -m virtualenv env
```

Comment:
Could use `venv`

Reply:
Copied this from the previous version, but will make the update.